File: Day3/day2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <set>
using namespace std;

int main() {
  string line;
  ifstream data ("data.txt");
  int sum = 0;

  if (data.is_open())
  {
    while (getline(data,line)) 
    {
      set<char> s;
      cout << line << "\n";
      string one = line.substr(0, line.length()/2);
      string two = line.substr(line.length()/2);
      for (int i = 0; i < one.length(); i++) {
        for (int j = 0; j < two.length(); j++) {
          if (one[i] == two[j]) {
            cout << one[i] << "\n";
            s.insert(one[i]);
            
          }
        }
      }
      for (auto& str: s)
      {
        cout << str << "\n";
      }

    }

    data.close();
  }
  else cout << "Unable to open file";
  cout << "Sum: " << sum;


  return 0;
}

File: Day2/day2.py
string = ""
f = open("data.txt", "r")
for x in f:
  string += x

score = 0

arr = string.split("\n")

m2 = {
    "X": 1,
    "Y": 2,
    "Z": 3
}
for game in arr:
    if len(game) > 2:
        if game[0] == "A":
            if game[2] == "X":
                score += 0
                score += 3
            elif game[2] == "Y":
                score += 3
                score += 1
            else:
                score += 6 
                score += 2
            
        elif game[0] == "B":
            if game[2] == "X":
                score += 0
                score += 1
            elif game[2] == "Y":
                score += 3
                score += 2
            else:
                score += 6
                score += 3
        else:
            if game[2] == "X":
                score += 0
                score += 2 
            elif game[2] == "Y":
                score += 3
                score += 3 
            else:
                score += 6
                score += 1


print(score)



